cmd/genji/shell/command.go

```diff
@@ -136,3 +142,182 @@ func displaySuggestions(in string) error {
+		if fc.IsPrimaryKey {
+			buf.WriteString(" PRIMARY KEY")
+		}
+
+		if fc.IsNotNull {
+			buf.WriteString(" NOT NULL")
+		}
+	}
+
+	// Fields constraints close parenthesis.
+	if len(fcs) > 0 {
+		buf.WriteString("\n);")
+	} else {
+		buf.WriteString(";")
+	}
+
+	// Print CREATE TABLE statement.
+	_, err = fmt.Fprintln(w, buf.String())
+	if err != nil {
+		return err
+	}
+	buf.Reset()
+
+	// Indexes statement.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indexes {
+		u := ""
+		if index.Opts.Unique {
+			u = " UNIQUE"
+		}
+
+		info := fmt.Sprintf("CREATE%s INDEX %s ON %s (%s);", u, index.Opts.IndexName, index.Opts.TableName,
+			index.Opts.Path)
+		_, err = fmt.Fprintln(w, info)
+		if err != nil {
+			return err
+		}
+	}
+
+	q := fmt.Sprintf("SELECT * FROM %s", t.Name())
+	res, err := tx.Query(q)
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+
+	// Inserts statements.
+	insert := fmt.Sprintf("INSERT INTO %s VALUES ", t.Name())
+	return res.Iterate(func(d document.Document) error {
+		buf.WriteString(insert)
+
+		data, err := document.MarshalJSON(d)
+		if err != nil {
+			return err
+		}
+		buf.Write(data)
+		buf.WriteString(";")
+
+		if _, err = fmt.Fprintln(w, buf.String()); err != nil {
+			return err
+		}
+
+		buf.Reset()
+
+		return nil
+	})
+}
+
+// runDumpCmd run .dump command where tables slice is the given tables or it can be empty that is consider all database.
+func runDumpCmd(db *genji.DB, tables []string, w io.Writer) error {
+	tx, err := db.Begin(false)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+	_, err = fmt.Fprintln(w, "BEGIN TRANSACTION;")
+	if err != nil {
+		return err
+	}
+
+	i := 0
+	for _, table := range tables {
+		err = dumpTable(tx, table, w)
+		switch err {
+		case nil:
+			// Blank separation between tables.
+			if i > 0 {
+				fmt.Println()
+			}
+			i++
+
+		case database.ErrTableNotFound: // If table doesn't exist we skip it.
+			continue
+		default:
+			_, err = fmt.Fprintln(w, "ROLLBACK;")
+			return err
+		}
+	}
+
+	// tables slice argument is not empty, all args tables has been displayed.
+	// If it is empty we should print the database content.
+	if len(tables) > 0 {
+		_, err = fmt.Fprintln(w, "COMMIT;")
+		return err
+	}
+
+	// tables slice argument is empty.
+	// Dump database content.
+	res, err := tx.Query("SELECT table_name FROM __genji_tables")
+	if err != nil {
+		_, err = fmt.Fprintln(w, "ROLLBACK;")
+		return err
+	}
+	defer res.Close()
+
+	i = 0
+	err = res.Iterate(func(d document.Document) error {
+		// Blank separation between tables.
+		if i > 0 {
+			fmt.Println()
+		}
+		i++
+
+		// Get table name.
+		var tableName string
+		if err := document.Scan(d, &tableName); err != nil {
+			return err
+		}
+
+		if err := dumpTable(tx, tableName, w); err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		_, err = fmt.Fprintln(w, "ROLLBACK;")
+		return err
+	}
+
+	_, err = fmt.Fprintln(w, "ROLLBACK;")
```

Comment:
```suggestion
	_, err = fmt.Fprintln(w, "COMMIT;")
```

Reply:
Even if it is not `ErrTableNotFound` we do a `COMMIT;`